### DISCUSSION TITLE
How to use llama.dll

### DISCUSSION THREAD

**Discussion by SevaSk** [ggml-org/llama.cpp#6984] [Q&A] [answered]
Sorry newbie question! I want to load llama.dll, I didn't compile it myself I just got it from the release page, but there is no .lib file. I don't really want to load every function pointer manually. Is there a better way?

**Comment by slaren** [comment, marked as answer]
You would need to build from source. You can use `cmake` to build or generate MSVC project files.

**Reply by SevaSk** [reply]
Thanks! Would it matter what cmake args I include? Would the .lib be the same?

**Reply by slaren** [reply]
I think so, the ABI should be the same regardless of the build args. You have to specify `-DBUILD_SHARED_LIBS=ON` to build as a dynamic library, though.

**Reply by SevaSk** [reply]
Great!
